alacritty_terminal/src/ansi.rs

```diff
@@ -1080,12 +1080,18 @@ where
             },
             ('r', None) => {
                 let arg0 = arg_or_default!(idx: 0, default: 1) as usize;
+                let arg1 = arg_or_default!(idx: 1, default: handler.lines().0 as _) as usize;
+
+                if arg0 >= arg1 {
+                    debug!("Invalid scroll region setting: ({};{})", arg0, arg1);
+                    return;
+                }
+
                 let top = Line(arg0 - 1);
                 // Bottom should be included in the range, but range end is not
-                // usually included.  One option would be to use an inclusive
+                // usually included. One option would be to use an inclusive
                 // range, but instead we just let the open range end be 1
                 // higher.
-                let arg1 = arg_or_default!(idx: 1, default: handler.lines().0 as _) as usize;
                 let bottom = Line(arg1);
```

Comment:
I think there's little reason to separate top/bottom like this. It looks a bit silly I think. The comment talks about both anyways, so it should be fine to do comment -> let top -> let bottom. Then everything's in a nice order and looks good.